draft-ietf-httpbis-message-signatures.md

```diff
@@ -357,6 +366,55 @@ The following example shows canonicalized values for different component identif
 
 Note that the value for `key="c"` has been re-serialized according to the strict `member_value` algorithm.
 
+
+### Binary-wrapped HTTP Fields {#http-header-byte-sequence}
+
+If the value of the the HTTP field in question is known by the application to cause problems with serialization, particularly with combination of multiple values as discussed in {{security-non-list}}, the signer MAY include the `bs` parameter in a component identifier to indicate the values of the fields need to be wrapped as binary structures before being combined.
+
+If this parameter is included with a component identifier, the component value is calculated using the following algorithm:
+
+0. Let the input be the ordered set of values for a field
+1. For each field value in the set:
+    0. Strip leading and trailing whitespace from each item in the list. Note that since HTTP field values are not allowed to contain leading and trailing whitespace, this will be a no-op in a compliant implementation.
+    1. Remove any obsolete line-folding within the line and replace it with a single space (" "), as discussed in {{Section 5.2 of HTTP1}}. Note that this behavior is specific to {{HTTP1}} and does not apply to other versions of the HTTP specification.
+    2. Encode the string as a Byte Sequence
```

Comment:
What does this mean? Maybe: encode the list as a structured field Byte Sequence using the list's ASCII representation. (This assumes that only ASCII is allowed, but that's really part of the problem.)

Reply:
@mnot is this an accurate way to say "take the header value as a binary"? To create the examples I encoded the strings, and field values are already a subset of ASCII characters allowed.

```suggestion
    2. Encode the field value's ASCII representation as a Byte Sequence
```